src/herder/Upgrades.cpp

```diff
@@ -66,8 +66,15 @@ Upgrades::Upgrades(UpgradeParameters const& params) : mParams(params)
 }
 
 void
-Upgrades::setParameters(UpgradeParameters const& params)
```

Comment:
I'm not sure about passing whole Config here as we only need one value of it. What about passing `uint32_t expectedProtocol` version instead?